### PR TITLE
🐛 Fix +required tag and update tests

### PR DESCRIPTION
**Pull request by damemi**
The `// +required` tag wasn't actually taking effect except for parent structs. This fixes it and updates test types to match

### CODE REVIEW

File: pkg/internal/codegen/parse/util.go [hunk 1]
```diff
@@ -375,7 +375,7 @@ func parseDescription(res []string) string {
 	var temp strings.Builder
 	var desc string
 	for _, comment := range res {
-		if !(strings.Contains(comment, "+kubebuilder") || strings.Contains(comment, "+optional")) {
+		if !(strings.Contains(comment, "+kubebuilder") || strings.Contains(comment, "+optional") || strings.Contains(comment, "+required")) {
```
Comment (damemi) [inline pkg/internal/codegen/parse/util.go:378]: @sttts we'll also need to add an exception for `+nullable` or take a better approach than OR'ing every tag here. Maybe we could assume if a line begins with `+` it's a tag and not to include it in the description?

File: pkg/internal/codegen/parse/util.go [hunk 2]
```diff
@@ -233,8 +233,8 @@ func hasSingular(t *types.Type) bool {
 
 // hasRequired returns true if t is annotated with
 // +required
-func hasRequired(t *types.Type) bool {
-	for _, c := range append(append([]string{}, t.SecondClosestCommentLines...), t.CommentLines...) {
+func hasRequired(t types.Member) bool {
+	for _, c := range t.CommentLines {
```
Comment (sttts) [inline pkg/internal/codegen/parse/util.go:237]: intentional?

Comment (damemi) [reply pkg/internal/codegen/parse/util.go:237]: @sttts yeah, `Member`s of a `Type` don't have `SecondClosestCommentLines`

Comment (sttts) [reply pkg/internal/codegen/parse/util.go:237]: Interesting. But looks like that.

**Pull request merged**